shuttle/src/sync/atomic/mod.rs

```diff
@@ -235,4 +240,75 @@ impl<T: Copy + Eq> Atomic<T> {
             s.update_clock(self_clock.as_ref().unwrap());
         });
     }
+
+    #[cfg(test)]
+    fn signature(&self) -> ResourceSignature {
+        self.signature.clone()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::sync::atomic::*;
+
+    #[test]
+    fn unique_resource_signatures() {
+        let atomic_i8 = AtomicI8::new(0);
+        let atomic_i16 = AtomicI16::new(0);
+        let atomic_i32 = AtomicI32::new(0);
+        let atomic_i64 = AtomicI64::new(0);
+        let atomic_isize = AtomicIsize::new(0);
+        let atomic_u8 = AtomicU8::new(0);
+        let atomic_u16 = AtomicU16::new(0);
+        let atomic_u32 = AtomicU32::new(0);
+        let atomic_u64 = AtomicU64::new(0);
+        let atomic_usize = AtomicUsize::new(0);
+        let atomic_bool = AtomicBool::new(false);
+        let atomic_ptr = AtomicPtr::new(std::ptr::null_mut::<i32>());
+
+        // All atomics should have unique signatures
+        let signatures = vec![
+            atomic_i8.signature(),
+            atomic_i16.signature(),
+            atomic_i32.signature(),
+            atomic_i64.signature(),
+            atomic_isize.signature(),
+            atomic_u8.signature(),
+            atomic_u16.signature(),
+            atomic_u32.signature(),
+            atomic_u64.signature(),
+            atomic_usize.signature(),
+            atomic_bool.signature(),
+            atomic_ptr.signature(),
+        ];
+
+        // Check all signatures are unique
+        assert_eq!(
+            signatures.len(),
+            signatures.iter().collect::<std::collections::HashSet<_>>().len()
+        );
+    }
+
+    #[test]
+    fn atomic_signatures_consistent_across_shuttle_iterations() {
+        use std::collections::HashSet;
+        use std::sync::{Arc, Mutex};
+
+        let all_signatures = Arc::new(Mutex::new(HashSet::new()));
+        let all_signatures_clone = all_signatures.clone();
+
+        crate::check_random(
+            move || {
+                let atomic1 = AtomicBool::new(false);
+                let atomic2 = AtomicBool::new(true);
+
+                all_signatures_clone.lock().unwrap().insert(atomic1.signature());
+                all_signatures_clone.lock().unwrap().insert(atomic2.signature());
+            },
+            10,
+        );
+
+        // Should have exactly 2 unique signatures across all iterations
```

Comment:
Could do a stronger check by loading the atomic and distinguishing between them since they carry two different boolean values